cmake/information_helpers.cmake

```diff
@@ -1,11 +1,24 @@
+macro(ginkgo_interface_libraries_recursively INTERFACE_LIBS)
+    foreach(_libs ${INTERFACE_LIBS})
+        if (TARGET ${_libs})
+            list(APPEND GINKGO_INTERFACE_LIBS_FOUND "-l${_libs}")
+            get_target_property(GINKGO_LIBS_INTERFACE_LIBS "${_libs}" INTERFACE_LINK_LIBRARIES)
+            ginkgo_interface_libraries_recursively("${GINKGO_LIBS_INTERFACE_LIBS}")
+        endif()
+    endforeach()
+endmacro()
+
 macro(ginkgo_interface_information)
     get_target_property(GINKGO_INTERFACE_LINK_LIBRARIES ginkgo INTERFACE_LINK_LIBRARIES)
     set(GINKGO_INTERFACE_LINK_FLAGS "-L${CMAKE_INSTALL_PREFIX}/${GINKGO_INSTALL_LIBRARY_DIR} -lginkgo")
     set(GINKGO_INTERFACE_CXX_FLAGS "-I${CMAKE_INSTALL_PREFIX}/${GINKGO_INSTALL_INCLUDE_DIR}")
```

Comment:
We may also need to propagate include paths here (hwloc, MPI, ...) to be future-proof?

Reply:
Technically it's, for now, the same directory as we install hwloc if it wasn't present in the user environment/config settings. I guess that can be done as well?

About the linking order, for now I have to say it's fairly random. I believe the recursion order should indeed take care of that, but I wouldn't say there is any guarantee here.